File: avalon-modules/avalon-authorization/src/test/java/io/github/lishangbu/avalon/authorization/repository/OauthRegisteredClientRepositoryTest.java
package io.github.lishangbu.avalon.authorization.repository;

import io.github.lishangbu.avalon.authorization.entity.OauthRegisteredClient;
import jakarta.annotation.Resource;
import java.util.Optional;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/// OAuth 注册客户端 Repository 测试
///
/// 验证根据 clientId 查询客户端配置的能力，依赖 Liquibase 初始化的数据
///
/// @author lishangbu
/// @since 2025/8/20
class OauthRegisteredClientRepositoryTest extends AbstractRepositoryTest {

  @Resource private Oauth2RegisteredClientRepository oauth2RegisteredClientRepository;

  @Test
  void testFindByClientId() {
    Optional<OauthRegisteredClient> clientOptional =
        oauth2RegisteredClientRepository.findByClientId("client");
    Assertions.assertTrue(clientOptional.isPresent());
    OauthRegisteredClient oauthRegisteredClient = clientOptional.get();
    Assertions.assertEquals("1", oauthRegisteredClient.getId());
    Assertions.assertEquals("client", oauthRegisteredClient.getClientId());
    Assertions.assertEquals("{noop}client", oauthRegisteredClient.getClientSecret());
  }
}

File: avalon-modules/avalon-authorization/src/test/kotlin/io/github/lishangbu/avalon/authorization/repository/OauthRegisteredClientRepositoryTest.kt
package io.github.lishangbu.avalon.authorization.repository

import io.github.lishangbu.avalon.authorization.entity.dto.OauthRegisteredClientSpecification
import jakarta.annotation.Resource
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Test
import org.springframework.data.domain.PageRequest

/**
 * OAuth 注册客户端 Repository 测试
 *
 * 验证根据 clientId 查询客户端配置的能力，依赖数据库初始化的数据
 *
 * @author lishangbu
 * @since 2025/8/20
 */
class OauthRegisteredClientRepositoryTest : AbstractRepositoryTest() {
    @Resource
    private lateinit var oauth2RegisteredClientRepository: Oauth2RegisteredClientRepository

    @Test
    fun testFindByClientId() {
        val oauthRegisteredClient =
            requireNotNull(oauth2RegisteredClientRepository.findByClientId("client"))
        assertEquals("1", oauthRegisteredClient.id)
        assertEquals("client", oauthRegisteredClient.clientId)
        assertEquals("{noop}client", oauthRegisteredClient.clientSecret)
    }

    @Test
    fun testFindPage() {
        val page = oauth2RegisteredClientRepository.findAll(null, PageRequest.of(0, 10))
        assertEquals(2, page.totalRowCount)
        assertFalse(page.rows.isEmpty())
        assertEquals("1", page.rows.first().id)
    }

    @Test
    fun testFindPageWithEmptySpecification() {
        val page =
            oauth2RegisteredClientRepository.findAll(
                OauthRegisteredClientSpecification(),
                PageRequest.of(0, 10),
            )
        assertEquals(2, page.totalRowCount)
        assertFalse(page.rows.isEmpty())
    }

    @Test
    fun testFindAllBySpecification() {
        val list = oauth2RegisteredClientRepository.findAll(OauthRegisteredClientSpecification(clientId = "client"))
        assertEquals(1, list.size)
        assertEquals("client", list.first().clientId)
    }
}
